### PR TITLE
chore: Deprecation for StoreFinderService and StoreDataService

### LINKED ISSUE #11093: Deprecation for StoreFinderService and StoreDataService
In `StoreFinderService`,
Remove constructor deprecation and make `platformId` required.

In `StoreDataService`,
Remove constructor deprecation and make `winRef` required.

### PR DESCRIPTION
**Pull request by Zeyber**
Closes: #11093 #11441

### CODE REVIEW

**Review by ren-tham — COMMENTED** [review]
All changes/migration files look good. The build seems to be failing though, please have a look.

File: docs/migration/4_0.md [hunk 1]
```diff
@@ -175,6 +175,43 @@ So let's migrate to the new structure step by step.
...
 ## Breaking Changes Introduced in 4.0
 
+### StoreFinderService
+
+- Added `platformId` injection to constructor.
+- Methods `getStoreLatitude()` and `getStoreLongitude()` have been moved to this service from removed `StoreDataService`.
+
+### StoreDataService
+
+- Service has been removed and functions moved to `StoreFinderService`.
```
Comment (marlass) [inline docs/migration/4_0.md:215]: Can we have removed public api migration for that?

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/store-finder-list.component.migration.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export const STORE_FINDER_LIST_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  // feature-libs/storefinder/components/store-finder-search-result/store-finder-list/store-finder-list.component.ts
+  class: STORE_FINDER_LIST_COMPONENT,
+  importPath: SPARTACUS_STOREFINDER,
+  deprecatedParams: [
+    {
+      className: STORE_DATA_SERVICE,
+      importPath: SPARTACUS_STOREFINDER,
+    },
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/store-finder-list.component.migration.ts:21]: Missing param for Document.

Syntax like this is needed:

```
{
      className: DOCUMENT,
      literalInference: ANY_TYPE,
      injectionToken: {
        token: DOCUMENT_STRING,
        importPath: ANGULAR_COMMON,
      },
    },
    ```

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/google-map-renderer.service.migration.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      className: STORE_DATA_SERVICE,
+      importPath: SPARTACUS_STOREFINDER,
+    },
+  ],
+  addParams: [
+    {
+      className: STORE_FINDER_SERVICE,
+      importPath: SPARTACUS_STOREFINDER,
+    },
+  ],
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/google-map-renderer.service.migration.ts]: To make it produce correct order you also have to remove script loader and then add it again.

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/store-finder-list.component.migration.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  ],
+  addParams: [
+    {
+      className: STORE_FINDER_SERVICE,
+      importPath: SPARTACUS_STOREFINDER,
+    },
+  ],
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/store-finder-list.component.migration.ts]: To have correct order after migration we also need to remove the `document` and add it once again

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**